src/Mvc/Mvc.Core/src/FromBodyAttribute.cs

```diff
@@ -10,9 +10,27 @@ namespace Microsoft.AspNetCore.Mvc
     /// Specifies that a parameter or property should be bound using the request body.
     /// </summary>
     [AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
-    public class FromBodyAttribute : Attribute, IBindingSourceMetadata
+    public class FromBodyAttribute : Attribute, IBindingSourceMetadata, IAllowEmptyInputInBodyModelBinding
     {
+        private bool? _allowEmptyInputInBodyModelBinding;
+
         /// <inheritdoc />
         public BindingSource BindingSource => BindingSource.Body;
+
+        /// <summary>
+        /// Gets or sets the flag which decides whether body model binding (for example, on an
+        /// action method parameter with <see cref="FromBodyAttribute"/>) should treat empty
+        /// input as valid. <see langword="null"/> by default.
+        /// </summary>
+        /// <remarks>
+        /// When configured, takes precedence over <see cref="MvcOptions.AllowEmptyInputInBodyModelBinding"/>.
+        /// </remarks>
+        public bool AllowEmptyInputInBodyModelBinding
```

Comment:
Suggest the property in `IAllowEmptyInputInBodyModelBinding` should be non-nullable too. I don't see a use case for a tri-state property here.

Reply:
I was looking at being able to distinguish these two scenario:

```C#
.AddControllers(options => options.AllowEmptyInputInBodyModelBinding = true);
...

public IActionResult Post([FromBody(AllowEmptyInputInBodyModelBinding = false)] /* this API always require a body */ LoginModel model)

public IActionResult Post([FromBody] /* I'm ok with whatever was configured in MvcOptions */ CommentModel model)
```

Reply:
Then this property should be of type `bool?`

Reply:
Unfortunately attributes cannot have nullable properties that can be assigned to: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/attributes#attribute-parameter-types. We could create an enum type to represent it. While it might be more correct to have an explicit distinction between fallback to 'system default' vs 'explicitly false', users don't really query the attribute so it seems like overkill.

Reply:
Perhaps overkill but the tri-state approach you want won't work as long as the only `IAllowEmptyInputInBodyModelBinding` implementation has a non-nullable `bool`.

Reply:
The interface `IAllowEmptyInputInBodyModelBinding` is non-public. It's largely a result of weird layering - `BindingInfo` is in `Abstractions`, `FromBody` is in Mvc.Core. We're free to assign it the shape we think works best.

Reply:
`[FromBody(AllowEmpty = AllowEmptyBodyBehavior.Default)]`
`[FromBody(AllowEmpty = AllowEmptyBodyBehavior.Never)]`
`[FromBody(AllowEmpty = AllowEmptyBodyBehavior.Allow)]`

```
enum AllowEmptyBodyBehavior
{
  Default,
  Never,
  Allow
}